### PR TITLE
Add NOMADS for downloading data, update to 37 pressure levels, update solar radiation calc, hera configs

### PR DESCRIPTION
**Pull request by SadeghTabas-NOAA**
## Description
This PR provides updates on the `gdas_utility.py` script including:

- Module to get the GDAS data from `NOMADS` which is up to date and can be used for generating real-time GraphCast forecasts. 
- It provides the functionality to run the GraphCast model on `hera machine`.
- Option to generate data on 37 pressure levels as well as 13 pressure levels.
- Removing PySolar package for calc of `tisr` as it would be calculated inside GraphCast.

### Associated GraphCast Issue to close
- Closes https://github.com/NOAA-EMC/graphcast/issues/16
- Closes https://github.com/NOAA-EMC/graphcast/issues/17

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [ ] New input data.
  - [x] Updated input data.

### CODE REVIEW

File: NCEP/gdas_utility.py [hunk 1]
```diff
@@ -15,25 +16,27 @@
...
-from joblib import Parallel, delayed
-import pytz
-
+import requests
+from bs4 import BeautifulSoup
 
 
 class GFSDataProcessor:
-    def __init__(self, start_datetime, end_datetime, output_directory=None, download_directory=None, keep_downloaded_data=True):
+    def __init__(self, start_datetime, end_datetime, num_pressure_levels=13, download_source='nomads', output_directory=None, download_directory=None, keep_downloaded_data=True):
```
Comment (junwang-noaa) [inline NCEP/gdas_utility.py:24]: I want to confirm, the default "download_source" is nomads?

File: NCEP/gdas_utility.py [hunk 2]
```diff
@@ -44,7 +47,57 @@ def __init__(self, start_datetime, end_datetime, output_directory=None, download
...
+                    # Define the local file path
+                    local_file_path = os.path.join(local_directory, os.path.basename(obj_key))
+
+                    # Download the file from S3 to the local path
+                    self.s3.download_file(self.bucket_name, obj_key, local_file_path)
+                    print(f"Downloaded {obj_key} to {local_file_path}")
+    
+    def nomads(self, date_str, time_str, local_directory):
+        # Construct the URL for the data directory
+        gdas_url = f"https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/{self.root_directory}.{date_str}/{time_str}/atmos/"
```
Comment (junwang-noaa) [inline NCEP/gdas_utility.py:72]: What is "f" here before the url?

File: NCEP/gdas_utility.py (hunk 1 above)
Comment (SadeghTabas-NOAA) [reply NCEP/gdas_utility.py:24]: yes

File: NCEP/gdas_utility.py (hunk 2 above)
Comment (SadeghTabas-NOAA) [reply NCEP/gdas_utility.py:72]: in python it means these `{self.root_directory}.{date_str}/{time_str}` should be imputed and these are not strings, it tells python we have variables in the string

File: NCEP/graphcast_job_cloud.sh [hunk 3]
```diff
@@ -55,7 +55,7 @@ echo "Execution time for gdas_utility.py: $execution_time seconds"
 start_time=$(date +%s)
 echo "start runing graphcast to get real time 10-days forecasts for: $curr_datetime"
 # Run another Python script
-python3 run_graphcast.py -i source-gdas_date-"$curr_datetime"_res-0.25_levels-13_steps-40.nc -o "$curr_datetime"
+python3 run_graphcast.py -i source-gdas_date-"$curr_datetime"_res-0.25_levels-13_steps-1.nc -o "$curr_datetime"
```
Comment (junwang-noaa) [inline NCEP/graphcast_job_cloud.sh:58]: Will this run graphcast 1 step? The message (line 56) said the it's 10 day dorecast

File: NCEP/graphcast_job_hera.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+python3 gdas_utility.py "$prev_datetime" "$curr_datetime" -k no
+
+end_time=$(date +%s)  # Record the end time in seconds since the epoch
+
+# Calculate and print the execution time
+execution_time=$((end_time - start_time))
+echo "Execution time for gdas_utility.py: $execution_time seconds"
+
+start_time=$(date +%s)
+echo "start runing graphcast to get real time 10-days forecasts for: $curr_datetime"
```
Comment (junwang-noaa) [inline NCEP/graphcast_job_hera.sh:53]: Same here

File: NCEP/graphcast_job_cloud.sh (hunk 3 above)
Comment (SadeghTabas-NOAA) [reply NCEP/graphcast_job_cloud.sh:58]: that should be 2 that shows the size of inputs, let me update it

Comment (SadeghTabas-NOAA) [reply NCEP/graphcast_job_cloud.sh]: Now updated!

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**